### PR TITLE
vpn: ship our own container image

### PR DESCRIPTION
**Pull request by burgerdev**
### Context

The container image used in the VPN chart should be reproducible and stable.  The resulting image should be handled similar to s3proxy: it's built as a part of the Constellation release process and then consumed from a Helm chart in our registry.

### Proposed change(s)

* Build an image with nix, extract a single layer tarball.
* Include the layer in a Bazel OCI rule.
* Integrate the image into the publishing process.

### Additional info

- [AB#3554](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3554)

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: nix/container/vpn/strongswan.sh [hunk 1]
```diff
@@ -1,6 +1,11 @@
 #!/bin/sh
 
-# The charon binary is not included in the PATH generated by nixery.dev, find it manually.
+set -eux
```
Comment (malt3) [inline nix/container/vpn/strongswan.sh:3]: Is the -x intended? Feel free to keep.

File: nix/container/vpn/vpn.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+in
+
+stdenv.mkDerivation {
+  name = "image";
+
+  src = image;
+
+  buildInputs = with pkgs; [ gnutar jq ];
+
+  # unpackPhase = "true";
```
Comment (malt3) [inline nix/container/vpn/vpn.nix:70]:
```suggestion
```

**Review by malt3 — APPROVED** [review]
LGTM

File: nix/container/vpn/strongswan.sh (hunk 1 above)
Comment (burgerdev) [reply nix/container/vpn/strongswan.sh:3]: Yes, `-x` is currently set in the deployment and only moves to the script here. I wanted to have some logs for what happens before charon is ready, and `-x` is the poor man's solution for that :)

**Pull request merged**